kernel/task/src/lib.rs

```diff
@@ -221,6 +221,16 @@ impl TaskRef {
             *self.0.exit_value_mailbox.lock() = Some(val);
             self.0.task.runstate().store(RunState::Exited);
 
+            // Synchronize with the acquire fence in `JoinableTaskRef::join()`,
+            // as we have just stored the exit value that `join()` will load.
+            fence(Ordering::Release);
```

Comment:
We don't need a fence here because we lock (and unlock the mutex via `drop`), which has all the necessary fences to sync the data stored in the mutex.

Reply:
yes, for now we do. I should've written a note about this, but I was looking ahead to the future where i might do something without a mutex. But thinking deeper, we'd probably always use one.